tracks/haskell.sh

```diff
@@ -1,10 +1,33 @@
+function _override_stack_resolver() {
+    if [[ -z "$XR_HASKELL_STACK_RESOLVER" ]]; then
+        return
+    fi
+
+    # See https://docs.haskellstack.org/en/stable/pantry/#snapshot-location for resolver syntax.
+    # For the purpose of running exercises it's resonable just to allow lts and nightly.
+    if ! [[ $XR_HASKELL_STACK_RESOLVER =~ '^(lts-[0-9\.]+|nightly-[0-9-]+)$' ]]; then
```

Comment:
```suggestion
    if ! [[ "$XR_HASKELL_STACK_RESOLVER" =~ ^(lts-[0-9\.]+|nightly-[0-9-]+)$ ]]; then
```

As per #9, the right-hand side regular expression should not be quoted for Bash. I think quoting the left-hand side operand is safer.